core/DataAccess/LogQueryBuilder/JoinGenerator.php

```diff
@@ -53,6 +53,13 @@ private function addMissingTablesNeededForJoins()
             if (!$logTable->getColumnToJoinOnIdVisit()) {
                 $tableNameToJoin = $logTable->getLinkTableToBeAbleToJoinOnVisit();
 
+                if (empty($tableNameToJoin) && $logTable->getWaysToJoinToOtherLogTables()) {
+                    foreach ($logTable->getWaysToJoinToOtherLogTables() as $otherLogTable => $column) {
+                        $this->tables->hasJoinedTable($otherLogTable) || $this->tables->addTableToJoin($otherLogTable);
```

Comment:
can we use a regular if statement for better readability?

Reply:
sure